app.rb

```diff
@@ -0,0 +1,116 @@
+require_relative 'book'
+require_relative 'student'
+require_relative 'rental'
+require_relative 'teacher'
+
+class App
+  def initialize(parent)
+    @parent = parent
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def list_of_people
+    if @people.empty?
+      puts 'List Of People is empty'
+    else
+      puts 'List of People:'
+      @people.each do |person|
+        puts "[#{person.class}] NAME: #{person.name}, ID: #{person.id}, AGE: #{person.age}"
+      end
+    end
+    @parent.show_menu
+  end
+
+  def list_of_books
+    if @books.empty?
+      puts 'No Book In The List.'
+    else
+      puts 'List of Books available.'
+      @books.each { |book| puts "Title: #{book.title} By #{book.author}" }
+    end
+    @parent.show_menu
+  end
+
+  def create_person
+    puts 'Do you want to create a Student(1) or Teacher(2)? [ENTER NUMBER]:'
+    person_input = gets.chomp
+    case person_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Please add a valid input!'
+    end
+    @parent.show_menu
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+    print 'Age: '
+    age = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people.push(Teacher.new(specialization, age, name))
+    puts
+    puts 'Teacher is successfully created'
+    @parent.show_menu
+  end
+
```

Comment:
- [ ] Good job so far, in your app it is not possible to successfully create a new Student or Person, it would be nice to fix this error  ⬇️ 
![Screenshot_203](https://github.com/stino-x/PST-Classes/assets/69638013/73723597-8348-4c1b-9049-718f22fe9ce8)